src/app/add-item/add-item.component.ts

```diff
@@ -52,15 +50,21 @@ export class AddItemComponent implements OnInit {
       });
   }
 
-  private formNewActionItem(): ActionItem {
-    const projectNameValue = this.dialogForm.get('project').value;
-    const dueDateValue = this.dialogForm.get('dueDate').value;
-    const newItem: ActionItem = {
-      title: this.dialogForm.get('name').value,
-      projectName: projectNameValue.name,
+  private formNewActionItem(): AddActionItem {
+    const title = this.dialogForm.get('name');
+    const projectName = this.dialogForm.get('project');
+    const dueDate = this.dialogForm.get('dueDate');
+
+    if (!title || !projectName || !dueDate) {
+      throw 'Invalid Action Item data';
```

Comment:
**FIX:** I don't see test for that. Please make sure that you have code coverage at 100%.